src/app/home-page/widget/organizations/organizations.component.ts

```diff
@@ -1,55 +1,37 @@
-import { Component, OnInit } from '@angular/core';
-import { Observable, Subject } from 'rxjs';
+import { Component } from '@angular/core';
 import { UserQuery } from 'app/shared/user/user.query';
 import { DefaultService } from 'app/shared/openapi';
 import { OrganizationUpdateTime } from 'app/shared/openapi/model/organizationUpdateTime';
 import { debounceTime, takeUntil } from 'rxjs/operators';
 import { formInputDebounceTime } from 'app/shared/constants';
+import { FilteredList } from '../filtered-list';
 
 @Component({
   selector: 'app-organizations',
   templateUrl: './organizations.component.html',
   styleUrls: ['./organizations.component.scss']
 })
-export class OrganizationsComponent implements OnInit {
-  public myOrganizations;
-  public organizationFilterText;
-  public hasOrganizations = false;
-  public firstCall = true;
-  userId$: Observable<number>;
-  protected ngUnsubscribe: Subject<{}> = new Subject();
-
-  constructor(private userQuery: UserQuery, private defaultService: DefaultService) {}
-
-  ngOnInit() {
-    this.userId$ = this.userQuery.userId$;
-    this.userQuery.userId$.subscribe((userId: number) => {
-      this.getMyOrganizations();
-    });
+export class OrganizationsComponent extends FilteredList {
+  constructor(userQuery: UserQuery, defaultService: DefaultService) {
+    super(userQuery, defaultService);
   }
 
-  getMyOrganizations(): void {
+  getMyList(): void {
     this.defaultService
-      .getUserDockstoreOrganizations(10, this.organizationFilterText)
+      .getUserDockstoreOrganizations(10, this.filterText)
       .pipe(
         debounceTime(formInputDebounceTime),
         takeUntil(this.ngUnsubscribe)
```

Comment:
I don't think you need this, although it won't  hurt... HTTP client requests (which I assume this is) are finite observables, which you don't need to unsubscribe from (although @garyluu should correct me if I'm wrong)

Reply:
Gary said it should be ok.

Reply:
https://oicr.slack.com/archives/C16ET3CF4/p1552941626265900

Reply:
Hmm, I'd overlooked that when you posted.